src/main/kotlin/solve/scene/view/PlaneView.kt

```diff
@@ -139,17 +176,101 @@ class PlaneView(
+    }
+
+    private fun getColorWithOpacity() = plane.layerSettings.getColorWithOpacity(plane)
+
+    private fun setPlaneElementColor(newColor: Color) {
+        planeElement.color = newColor
+        addToFrameDrawer()
     }
 
-    private fun getColorWithOpacity(): Color {
-        val rgb = plane.layerSettings.getColor(plane)
-        return Color(rgb.red, rgb.green, rgb.blue, plane.layerSettings.opacity)
+    private fun redrawPlaneWithColor(newColor: Color) {
+        setPlaneElementColor(newColor)
+        frameDrawer.redrawPoints(planeElement.points)
+    }
+
+    private fun getDisabledPlaneColor() = getColorWithOpacity().withReplacedOpacity(LayerSettings.MinOpacity)
+
+    private fun onEnabledChanged() {
+        lateinit var planeVisibilityColor: Color
+
+        if (plane.layerSettings.enabled) {
+            planeVisibilityColor = lastEnabledPlaneElementColor
+            planeUIDLabel?.isVisible = true
+        } else {
+            planeVisibilityColor = Color.TRANSPARENT
+            planeUIDLabel?.isVisible = false
+        }
+
+        redrawPlaneWithColor(planeVisibilityColor)
+    }
+
+    private fun createUIDLabel(): Label {
+        val uidLabel = Label(plane.uid.toString())
+        uidLabel.textFill = getBlackOrWhiteContrastingTo(plane.layerSettings.getColor(plane))
+        uidLabel.font = Font.font(null, FontWeight.BOLD, UIDLabelFontSize)
+
+        return uidLabel
+    }
+
+    private fun calculatePlaneCenterPoint(): DoublePoint {
+        val centroidPoint = DoublePoint(plane.points.map { it.x }.average(), plane.points.map { it.y }.average())
+        val nearestToCentroid = plane.points.minBy { point ->
+            DoublePoint(point.x.toDouble(), point.y.toDouble()).distanceTo(centroidPoint)
+        }
+
+        return DoublePoint(nearestToCentroid.x.toDouble(), nearestToCentroid.y.toDouble())
+    }
+
+    private fun updateUIDLabelPosition() {
+        node.layoutX = uidLabelCoordinates.x - (planeUIDLabel?.width ?: 0.0) / 2.0
+        node.layoutY = uidLabelCoordinates.y - (planeUIDLabel?.height ?: 0.0) / 2.0
+    }
+
+    private fun hideUIDLabel() {
+        if (!isShowingUIDLabel) {
+            return
+        }
+
+        node.clearChildren()
+        isShowingUIDLabel = false
+    }
+
+    private fun showUIDLabel() {
+        if (isShowingUIDLabel) {
+            return
+        }
+
+        if (planeCenterPoint == null) {
+            planeCenterPoint = calculatePlaneCenterPoint()
+        }
+        if (planeUIDLabel == null) {
+            planeUIDLabel = createUIDLabel()
+        }
+        val uidLabel = planeUIDLabel ?: return
+
+        uidLabel.isVisible = false
+        node.add(this.planeUIDLabel ?: return)
+
+        // Without delay the spawn coordinates of the uid label are incorrect.
+        val uidSpawnCoroutineScope = CoroutineScope(Dispatchers.JavaFx)
+        uidSpawnCoroutineScope.launch(Dispatchers.JavaFx) {
+            delay(PlaneUIDLabelSpawnDelayMillis)
```

Comment:
Why 25 mills is always enough?

Reply:
> Why 25 mills is always enough?

It is determined empirically

Reply:
No, 25 looks not enough. I've installed a 50 millis